lib/core/src/display/navigation/mouse_manager.rs

```diff
@@ -0,0 +1,295 @@
+use crate::display::rendering::DOMContainer;
+use crate::system::web::dyn_into;
+use crate::system::web::Result;
+use crate::system::web::Error;
+use crate::system::web::ignore_context_menu;
+
+use wasm_bindgen::prelude::*;
+use wasm_bindgen::JsCast;
+use web_sys::MouseEvent;
+use web_sys::WheelEvent;
+use web_sys::EventTarget;
+use web_sys::AddEventListenerOptions;
+use js_sys::Function;
+use nalgebra::Vector2;
+use std::rc::Rc;
+use std::cell::RefCell;
+
+// ======================
+// === EventListeners ===
+// ======================
+
+/// A mouse event listener object returned when adding callbacks to MouseManager.
+pub type MouseEventListener = Closure<dyn Fn(MouseEvent)>;
+/// A wheel event listener object returned when adding callbacks to MouseManager.
+pub type WheelEventListener = Closure<dyn Fn(WheelEvent)>;
+
+// ===================
+// === MouseButton ===
+// ===================
+
+/// An enumeration representing the mouse buttons.
+pub enum MouseButton {
+    LEFT,
+    MIDDLE,
+    RIGHT
+}
+
+// =======================
+// === MouseClickEvent ===
+// =======================
+
+/// A struct storing information about mouse down and mouse up events.
+pub struct MouseClickEvent {
+    pub position : Vector2<f32>,
+    pub button   : MouseButton
+}
+
+impl MouseClickEvent {
+    fn from(event:MouseEvent, data:&Rc<RefCell<MouseManagerData>>) -> Self {
+        let position  = Vector2::new(event.x() as f32, event.y() as f32);
+        let position  = position - data.borrow().dom.position();
+        let button    = match event.button() {
+            LEFT_MOUSE_BUTTON      => MouseButton::LEFT,
+            MIDDLE_MOUSE_BUTTON    => MouseButton::MIDDLE,
+            RIGHT_MOUSE_BUTTON | _ => MouseButton::RIGHT
```

Comment:
What if we would like to handle other buttons on mouse? We should not default to right mouse button.

Reply:
I think it's a good semantic for now. We can handle other buttons in the future.

Reply:
Why don't we just drop other events instead of converting them to RMB?